### ISSUE TITLE
Cannot use full period of time units (now/h) in alert condition

### ISSUE THREAD

**Issue by visit1985** [grafana/grafana#42296]
**What happened**:

We have a business requirement, to alert on the error rate in the last hour, for API calls stored in CloudWatch Logs.

We created a CloudWatch Logs query grouping values by full hours using [`bin(1h)`](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CWL_QuerySyntax.html#CWL_QuerySyntax-datetime-functions-table). Something along the lines of:

```
filter path="/api/v1/something" | stats sum(status_code > 299) / sum(status_code > 0) * 100 as `Error Rate` by bin(1h)
```

In order to get one single correct result, we need to run this query with a full hour time range (like `12:00:00` to `12:59:59`).

In the dashboards time range selector, we can specify `now/h-1h` as `From` and `To` to send the correct query to CloudWatch Logs.

But when we created an alert condition based on that:

```
WHEN max() OF query(Query, now/h-1h, now/h-1h) IS ABOVE 1
```

Testing this alert rule returns:

```
alert validation error: time: invalid duration "-now/h-1h"
```

**What you expected to happen**:

According to the [docs](https://grafana.com/docs/grafana/latest/dashboards/time-range-controls/#time-units-and-relative-ranges), we can append `/<time unit>` to a time range to get the full period of the unit (day, week, month, etc…).
We expect this to work in [query conditions](https://grafana.com/docs/grafana/latest/alerting/old-alerting/create-alerts/#query-condition-example) as well.

**Environment**:
- Grafana version: v8.2.1
- Data source type & version: CloudWatch
- OS Grafana is installed on: Docker on Linux
- User OS & Browser: Chrome

**Comment by zuchka** [comment]
Thanks for creating this issue, @visit1985

> We expect this to work in [query conditions](https://grafana.com/docs/grafana/latest/alerting/old-alerting/create-alerts/#query-condition-example) as well.

I noticed that the link above points to documentation for Legacy Grafana Alerts. Can you please verify:
* are you using legacy alerts on 8.2.1, or are you using the new Unified Alerting platform? 

Unified Alerting will replace legacy alerting:
* https://grafana.com/docs/grafana/latest/alerting/unified-alerting/
* https://grafana.com/blog/2021/06/14/the-new-unified-alerting-system-for-grafana-everything-you-need-to-know/
* https://grafana.com/go/webinar/intro-to-unified-alerting-in-grafana/

**Comment by visit1985** [comment]
Hey @zuchka,

in our first attempt we used legacy alerts. Now we retestet it on unified alerting, and cannot even set a time range like `now/h` because the field validation says `Value not in relative time format.`.


![image](https://user-images.githubusercontent.com/2237222/143858170-48b262bc-e031-4e84-8fb4-cbfdda3eba11.png)

**Comment by davidmparrott** [comment]
@visit1985 for the unified alerting condition creation aspect, it looks like we're using a slightly different syntax than what is documented. Try `now-1h` to `now` and you should stop having the validation error. We also appear to not support `M` but we do support `d` and you can use, for example, `now-30d` to get the last thirty days. 

That doesn't solve the problem of this being incorrectly documented for unified alerting and I thank you for pointing this out. Can you give my suggestion a try and see if that helps you get unblocked?

**Comment by visit1985** [comment]
@davidmparrott, thanks for your answer, but this issue is not about substracting time units. It‘s about truncating `now` to full hours.

**Comment by visit1985** [comment]
related #36154

**Comment by ivanahuckova** [comment]
@visit1985 I am going to close this as duplicate of  #36154 as there is more context and information on what and why is happening. Please check #36154 for progress on this issue. Thank for bringing this up! 🙏
